packages/react-components/react-motion-components-preview/library/src/components/Scale/Scale.ts

```diff
@@ -1,46 +1,64 @@
-import {
-  motionTokens,
-  PresenceMotionFn,
-  createPresenceComponent,
-  createPresenceComponentVariant,
-} from '@fluentui/react-motion';
+import { motionTokens, createPresenceComponent } from '@fluentui/react-motion';
+import { PresenceMotionFnCreator } from '../../types';
+import { opacityAtom, scaleAtom } from './ScaleAtoms';
+import { ScaleRuntimeParams_unstable, ScaleVariantParams_unstable } from './Scale.types';
 
 /** Define a presence motion for scale in/out */
-const scaleMotion: PresenceMotionFn<{ animateOpacity?: boolean }> = ({ animateOpacity = true }) => {
-  const fromOpacity = animateOpacity ? 0 : 1;
-  const toOpacity = 1;
-  const fromScale = 0.9; // Could be a custom param in the future
-  const toScale = 1;
-
-  const enterKeyframes = [
-    { opacity: fromOpacity, transform: `scale3d(${fromScale}, ${fromScale}, 1)`, visibility: 'visible' },
-    { opacity: toOpacity, transform: `scale3d(${toScale}, ${toScale}, 1)` },
-  ];
-
-  const exitKeyframes = [
-    { opacity: toOpacity, transform: `scale3d(${toScale}, ${toScale}, 1)` },
-    { opacity: fromOpacity, transform: `scale3d(${fromScale}, ${fromScale}, 1)`, visibility: 'hidden' },
-  ];
-
-  return {
-    enter: {
-      duration: motionTokens.durationGentle,
-      easing: motionTokens.curveDecelerateMax,
-      keyframes: enterKeyframes,
-    },
-    exit: { duration: motionTokens.durationNormal, easing: motionTokens.curveAccelerateMax, keyframes: exitKeyframes },
-  };
-};
+export const createScalePresence: PresenceMotionFnCreator<ScaleVariantParams_unstable, ScaleRuntimeParams_unstable> =
+  ({
+    enterDuration = motionTokens.durationGentle,
+    enterEasing = motionTokens.curveEasyEaseMax,
+    exitDuration = motionTokens.durationNormal,
+    exitEasing = motionTokens.curveAccelerateMax,
+  } = {}) =>
+  ({ animateOpacity = true }) => ({
+    enter: [
+      scaleAtom({
+        duration: enterDuration,
+        easing: enterEasing,
+      }),
+      ...(animateOpacity
+        ? [
+            opacityAtom({
+              duration: enterDuration,
+              easing: enterEasing,
+            }),
+          ]
+        : []),
+    ],
+    exit: [
+      scaleAtom({
+        duration: exitDuration,
+        easing: exitEasing,
+      }),
+      ...(animateOpacity
+        ? [
+            opacityAtom({
+              duration: exitDuration,
+              easing: exitEasing,
+            }),
+          ]
+        : []),
```

Comment:
I understand the thinking behind this refactoring and the functional style, and the motivations are fine. But with the resulting code, it's debatable as to whether the readability has become better or worse. The ternary spread is clever and functional, but it's verbose and not straightforward to understand like a simple imperative `if`.